cmd/manager/main.go

```diff
@@ -4,140 +4,152 @@ import (
+// TODO(tflannag): We should avoid having to maintain a global list of mappings for the default CatalogSource(s)
+// TODO(tflannag): What's the openshift recommendation on leader election? Do we want to avoid using the current leader for life implementation?
 
-var (
-	version     = flag.Bool("version", false, "displays marketplace source commit info.")
-	tlsKeyPath  = flag.String("tls-key", "", "Path to use for private key (requires tls-cert)")
-	tlsCertPath = flag.String("tls-cert", "", "Path to use for certificate (requires tls-key)")
+const (
+	leaderElectionConfigMapName = "marketplace-operator-lock"
 )
 
 func printVersion() {
-	log.Printf("Go Version: %s", runtime.Version())
-	log.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	log.Printf("operator-sdk Version: %v", sdkVersion.Version)
+	logrus.Printf("Go Version: %s", runtime.Version())
+	logrus.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+	logrus.Printf("operator-sdk Version: %v", sdkVersion.Version)
 }
 
 func main() {
 	printVersion()
 
-	// Parse the command line arguments
-	flag.StringVar(&defaults.Dir, "defaultsDir",
-		"", "the directory where the default CatalogSources are stored")
-	var clusterOperatorName string
+	var (
+		clusterOperatorName string
+		tlsKeyPath          string
+		tlsCertPath         string
+		version             bool
+	)
 	flag.StringVar(&clusterOperatorName, "clusterOperatorName", "", "the name of the OpenShift ClusterOperator that should reflect this operator's status, or the empty string to disable ClusterOperator updates")
+	flag.StringVar(&defaults.Dir, "defaultsDir", "", "the directory where the default CatalogSources are stored")
+	flag.BoolVar(&version, "version", false, "displays marketplace source commit info.")
+	flag.StringVar(&tlsKeyPath, "tls-key", "", "Path to use for private key (requires tls-cert)")
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to use for certificate (requires tls-key)")
 	flag.Parse()
 
 	// Check if version flag was set
-	if *version {
-		fmt.Print(sourceCommit.String())
-		// Exit immediately
+	if version {
+		logrus.Infof(sourceCommit.String())
 		os.Exit(0)
 	}
 
 	// set TLS to serve metrics over a secure channel if cert is provided
 	// cert is provided by default by the marketplace-trusted-ca volume mounted as part of the marketplace-operator deployment
-	var useTLS bool
-	if *tlsCertPath != "" && *tlsKeyPath == "" || *tlsCertPath == "" && *tlsKeyPath != "" {
-		log.Warn("both --tls-key and --tls-crt must be provided for TLS to be enabled, falling back to non-https")
-	} else if *tlsCertPath == "" && *tlsKeyPath == "" {
-		log.Info("TLS keys not set, using non-https for metrics")
-	} else {
-		log.Info("TLS keys set, using https for metrics")
-		useTLS = true
-	}
-	err := metrics.ServePrometheus(useTLS, *tlsCertPath, *tlsKeyPath)
+	err := metrics.ServePrometheus(tlsCertPath, tlsKeyPath)
```

Comment:
Another minor change: moved the `useTLS` logic in its own function in the metrics library and updated the `ServePrometheus` function to remove that `useTLS` parameter and instead call that helper function in its flow.

Reply:
Also, it looks like we don't have a test that ensures the Prometheus ServiceMonitor resource hasn't regressed, so we may need to call that out in a TODO comment. I verified we're still serving Prometheus metrics ourselves locally:

```bash
#! /bin/bash

set -o errexit
set -o nounset
set -o pipefail

token=$(oc -n openshift-monitoring sa get-token prometheus-k8s)
host=$(oc -n openshift-monitoring get routes prometheus-k8s -o jsonpath={.spec.host})

# Note on the usage:
# Curl is particularly annoying to pass variables to parameters, so in order to
# a query successfully you need to ensure that the `""` and `''` are properly formatted
# in order to avoid a parsing error. You either need to avoid using the same string
# deliminitor entirely or character escape them.
#
# The following shows this issue as an example:
# $ ./scripts/run-prom-query.sh 'rate(container_cpu_usage_seconds_total{image!="", container_name!="POD"}[5m])'
curl -k -s -G -H "Authorization: Bearer $token" --data-urlencode "query=$1" https://${host}/api/v1/query | faq -f json
```
```bash
$ ./scripts/run-prom-query.sh 'sum(kube_pod_container_resource_requests_cpu_cores{namespace="openshift-marketplace"}) by (pod, namespace, node)' | wc -l
63
```

Reply:
It looks like in the past we were carrying a PrometheusRule YAML manifest, but I didn't dig into whether we still care about serving our own custom metrics as it looks like they were mainly centered around (the now deprecated) app registry implementation.

When poking the Prometheus rules endpoint locally, it looks like we're not registering any custom rules anymore.

Reply:
Ah, good to know - either way, the infrastructure is still there we just need to update the metric registration handler. Just wanted to call out this behavior anyways.